tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -79,201 +73,50 @@ public AspNetCoreDiagnosticObserver(Tracer tracer, Security security)
 
         protected override string ListenerName => DiagnosticListenerName;
 
-        private Tracer CurrentTracer => _tracer ?? Tracer.Instance;
+        private Tracer CurrentTracer => _tracer ?? (_tracer = Tracer.Instance);
 
-        private Security CurrentSecurity => _security ?? Security.Instance;
+        private Security CurrentSecurity => _security ?? (_security = Security.Instance);
```

Comment:
I don't understand why making these fields non `readonly` is required or preferable 🤔